### PR TITLE
add named_axiskeys

### LINKED ISSUE #37: Axiskeys as a NamedTuple?
Is there a nice way to do the following:
```julia
julia> using AxisKeys

julia> arr = KeyedArray(randn(2,2), a=1:2, b=3:4);

julia> namedaxiskeys(arr) # is there such a function/how to do this?
(a=1:2, b=3:4)
```

### PR DESCRIPTION
**Pull request by jw3126**
close #37

### CODE REVIEW

File: src/AxisKeys.jl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module AxisKeys
 
 include("struct.jl")
-export KeyedArray, axiskeys
+export KeyedArray, axiskeys, named_axiskeys
```
Comment (jw3126) [inline src/AxisKeys.jl:4]: or spell it `namedaxiskeys`? Should it be exported?

File: src/struct.jl [hunk 2]
```diff
@@ -50,6 +50,26 @@ axiskeys(x::KeyedVector) = tuple(getindex(getfield(x, :keys)))
...
+
+julia> arr = KeyedArray([1 2], x=[1], y=[2,3]);
+
+julia> named_axiskeys(arr)
+(x = [1], y = [2, 3])
+```
+"""
+function named_axiskeys(arr)::NamedTuple
+    NT = NamedTuple{dimnames(arr)}
+    return NT(axiskeys(arr))
```
Comment (mcabbott) [inline src/struct.jl:70]:
```suggestion
function named_axiskeys(x)
    NT = NamedTuple{dimnames(x)}
    NT(keys_or_axes(x))
```
Not 100% sure, but with `keys_or_axes` it still returns something on a NamedDimsArray.

Comment (jw3126) [reply src/struct.jl:70]: Since `axiskeys` does not work on `NamedDimsArray`, it does seem consistent that `named_axiskeys` does not work either?

Comment (mcabbott) [reply src/struct.jl:70]: Yes that may be more logical.

File: src/AxisKeys.jl (hunk 1 above)
Comment (mcabbott) [reply src/AxisKeys.jl:4]: Name seems OK, it follows NamedDims's & is getting a bit long as one word. Could also be `named_keys` to shorten it. 

Would you mind moving it to the end of file names.jl?

Comment (jw3126) [reply src/AxisKeys.jl]: done.

File: src/struct.jl (hunk 2 above)
Comment (jw3126) [reply src/struct.jl]: Ok I will then leave it as is.

**Comment by mcabbott** [comment]
Thanks! 

And good catch on the `propertynames` thing.

**Pull request merged**